### ISSUE TITLE
How do you set it up correctly?

### ISSUE THREAD

**Issue by Magisdn** [ravenscroftj/freshrss-flaresolverr-extension#6]
Hi!

FreshRSS Flaresolverr Extension installed using docker compose
```
version: "2.1"
services:
  flaresolverr:     
    image: ghcr.io/flaresolverr/flaresolverr:latest
    restart: always
    environment:
      - LOG_LEVEL=info
    ports:
      - 8191:8191

  freshrss:
    image: lscr.io/linuxserver/freshrss:latest
    container_name: freshrss
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - ./data:/config
    ports:
      - 9089:80
    restart: unless-stopped
```
in 
Please enter the URL of your FlareSolverr instance
 http://192.168.1.7:8191

Prepend your feed URL with: http://192.168.1.7/api/cloudsolver.php?feed=

In Freshrss journal
cURL error 22: The requested URL returned error: 404 [http://192.168.1.7/api/cloudsolver.php?feed=https://sanet.st/books/tag/medical/rss/]

I open it in the browser

http://192.168.1.7:8191/

`{"msg": "FlareSolverr is ready!", "version": "3.3.10", "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"}
`
```
http://192.168.1.7:8191/api/cloudsolver.php?feed=https://www.gateworld.net/feed/
{"error": "Not found: '/api/cloudsolver.php'", "status_code": 404}
```

PS. You may have an error in the readme
"... and set the feed url in FreshRSS to: 
https://freshrss.example.com/api/cloudsolver.php?feed=[https:/](https://nlpnewsletter.substack.com/feed)https://nlpnewsletter.substack.com/feed"

**Comment by BiscuitHub** [comment]
Good day @Magisdn 

If you're still at this, the Prepend+FeedUrl should be:

`https://[your-freshrss-address]/api/cloudsolver.php?feed=`

so I'm guessing for your install it is:

`https://192.168.1.7:9089/api/cloudsolver.php?feed=https://www.gateworld.net/feed/`

**Comment by JoshuaArus** [comment]
Using docker is slightly different: 

remove the ports part of flaresolverr container (exposition of this port outside the containers is not mandatory) : 

```
version: "2.1"
services:
  flaresolverr:     
    image: ghcr.io/flaresolverr/flaresolverr:latest
    restart: always
    environment:
      - LOG_LEVEL=info

  freshrss:
    image: lscr.io/linuxserver/freshrss:latest
    container_name: freshrss
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - ./data:/config
    ports:
      - 9089:80
    restart: unless-stopped
```

Then, the URL to configure in the extension is not `http://192.168.1.7:8191/` but `http://flaresolverr:8191/`

**Comment by Nelluk** [comment]
I've been trying to get this to work and having trouble like the OP. I initially tried my docker compose similar to @JoshuaArus - with no port specified and http://flaresolverr:8191/ listed in the extension config, but that gets me a refusal to connect if I hit `http://192.168.1.244/api/cloudsolver.php?feed=https://site.example/rss`

I then updated it to the compose file shown in the README.md, which is more like OP has it initially, with the ports exposed and the config pointing to the flaresolverr instance running at 192.168.1.244:8191, but that gets the same results when I try to use it with the config.

**Comment by ravenscroftj** [comment]
@Nelluk can you share your compose file contents?

**Comment by Nelluk** [comment]
Sure, thanks @ravenscroftj 

```yaml
version: "2.4"
services:
  flaresolverr:      
    image: ghcr.io/flaresolverr/flaresolverr:latest
    restart: always
    environment:
      - LOG_LEVEL=info
    # ports:
    #   - 8191:8191
  freshrss:
    image: freshrss/freshrss:1
    container_name: freshrss
    restart: unless-stopped
    logging:
      options:
        max-size: 10m
    volumes:
      - ./data:/var/www/FreshRSS/data
      - ./extensions:/var/www/FreshRSS/extensions
    ports:
      - "8980:80"
    environment:
      TZ: America/Detroit
      CRON_MIN: '2,32'
      FRESHRSS_ENV: development
      LISTEN: 0.0.0.0:80
      TRUSTED_PROXY: 172.16.0.1/12 192.168.0.1/16
      OIDC_ENABLED: 0
      FRESHRSS_INSTALL: |-
        --api-enabled
        --base-url ${BASE_URL}
        --default_user admin
        --language en
      FRESHRSS_USER: |-
        --api-password ${ADMIN_API_PASSWORD}
        --email ${ADMIN_EMAIL}
        --language en
        --password ${ADMIN_PASSWORD}
        --user admin
```

**Comment by Nelluk** [comment]
Yes, 192.168.1.244 is my docker machine and I access FreshRSS at 192.168.1.244:8980. 
I get a 404 if I hit http://192.168.1.244:8980/api/cloudsolver.php?feed=https://site.example/rss

Can you clarify if the Flaresolverr should be configured with port 8191 and instance URL http://192.168.1.244:8191/ ? (I believe this is what the readme.md tells me). Or should I leave the port specification off the docker-compose and set the instance URL to http://flaresolverr:8191/ like this issue suggests? I get a 404 either way but not sure which of these two approaches I should be taking.

**Comment by ravenscroftj** [comment]
Try toggling the plugin off and on again from the admin panel - due to the way extensions work in docker, the cloudsolver.php file will get wiped and need to be recreated again (this usually happens after a `docker pull` application update).

WRT access via IP or container name, either is technically fine but I'd suggest using `flaresolverr:8191` if your freshrss and flaresolver images are inside the same compose file. This is generally a "nicer" approach since it keeps the containers encapsulated inside their own network and means you don't need to worry about other applications trying to use conflicting TCP ports or attackers trying to probe your infrastructure and making use of your flaresolverr instance (of course if your server is only accessible from a LAN then this is less of a concern anyway).

**Comment by Nelluk** [comment]
I've narrowed down the problem and this is a permissions issue. I was having the same problem as [issue 5](https://github.com/ravenscroftj/freshrss-flaresolverr-extension/issues/5) with a message of `Cannot enable extension FreshRss FlareSolverr: server cannot write in /var/www/FreshRSS/p/api`.   The suggested fix of `docker exec freshrss chown www-data:www-data -R /var/www/FreshRSS/p` works, but only as long as the container remains up.

So in this case I had done the fix before commenting on this thread, but had done a `docker compose restart` after a config tweak, and that reverted the permissions fix but this time with no error shown, until I tried your suggestion of toggling the plugin off and back on, at which point I get the same `Cannot enable extension` message.

To summarize, if I use the `docker exec` fix, and then turn off the plugin and back on, everything works. But if I ever restart or update FreshRSS I need to do the fix again.

This might be outside the scope of your help but do you know of a more permanent fix for this?

**Comment by ravenscroftj** [comment]
I'm glad you got it working. The permissions problem is a tricky one - it's a culmination of freshrss, docker, linuxserver (who make the freshrss docker image) and my plugin all interacting together in an annoying way.

In future I will see if I can hook more directly into the FreshRSS core application rather than providing my own php file.

I've opened #15 to track this problem

**Comment by philtr** [comment]
I was able to get this working by upgrading `docker compose` to > 2.30.0 — there is a new [`post_start` lifecycle hook](https://docs.docker.com/compose/how-tos/lifecycle/#post-start-hooks) that you can [use to modify the file ownership](https://github.com/philtr/frst/commit/ad94e01f71bac81b254071549ab1fca8506a94b1) after the container boots.
